benchmarks/benches/utils.rs

```diff
@@ -0,0 +1,121 @@
+use std::fs::{create_dir_all, remove_dir_all, File};
+
+use criterion::BenchmarkId;
+use heed::EnvOpenOptions;
+use milli::{
+    update::{IndexDocumentsMethod, Settings, UpdateBuilder, UpdateFormat},
+    FacetCondition, Index,
+};
+
+pub struct Conf<'a> {
+    /// where we are going to create our database.mmdb directory
+    /// each benchmark will first try to delete it and then recreate it
+    pub database_name: &'a str,
+    /// the dataset to be used, it must be an uncompressed csv
+    pub dataset: &'a str,
+    pub group_name: &'a str,
+    pub queries: &'a [&'a str],
+    /// here you can change which criterion are used and in which order.
+    /// - if you specify something all the base configuration will be thrown out
+    /// - if you don't specify anything (None) the default configuration will be kept
+    pub criterion: Option<&'a [&'a str]>,
+    /// the last chance to configure your database as you want
+    pub configure: fn(&mut Settings),
+    pub facet_condition: Option<&'a str>,
+    /// enable or disable the optional words on the query
+    pub optional_words: bool,
+    /// primary key, if there is None we'll auto-generate docids for every documents
+    pub primary_key: Option<&'a str>,
+}
+
+impl Conf<'_> {
+    fn nop(_builder: &mut Settings) {}
```

Comment:
What's that `nop` method for?

Reply:
I use it in my `BASE` configuration below.
I could've used an `Option` in the structure, but a no-operation also seems good to me and makes the code a little bit more straightforward.
Let me know what you think of it. 🙂

Reply:
Ho OK, couldn't you just use `drop`, instead? that's also a noop function.